src/api/environment.cc

```diff
@@ -242,9 +242,11 @@ void SetIsolateMiscHandlers(v8::Isolate* isolate, const IsolateSettings& s) {
     s.allow_wasm_code_generation_callback : AllowWasmCodeGenerationCallback;
   isolate->SetAllowWasmCodeGenerationCallback(allow_wasm_codegen_cb);
 
-  auto* promise_reject_cb = s.promise_reject_callback ?
-    s.promise_reject_callback : task_queue::PromiseRejectCallback;
-  isolate->SetPromiseRejectCallback(promise_reject_cb);
+  if ((s.flags & SHOULD_NOT_SET_PROMISE_REJECTION_CALLBACK) != 0) {
```

Comment:
```suggestion
  if ((s.flags & SHOULD_NOT_SET_PROMISE_REJECTION_CALLBACK) == 0) {
```

I think this might also be inverted?

Reply:
🤔 the check is that the flag is _not_ set, right? so we're checking that the flag is not present (e.g setting callback by default), which i think is a check for the that expression evaling to false?

Reply:
Currently, the `if` branch is taken if `SHOULD_NOT_SET_PROMISE_REJECTION_CALLBACK` is set, but we want to take it when `SHOULD_NOT_SET_PROMISE_REJECTION_CALLBACK` is not set, right?

Reply:
if the flag is present it means the callback should not be set. if the flag is not present it means the callback should be set.

Reply:
oh yes you're right! fixing, misparsed.